src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+    private final int figureCount = 5;
+    private final int defalteSide = 10;
+
+    public Figure getFigure() {
+        switch (getRandomFigure()) {
+            case 0:
+                return createCircle();
+            case 1:
+                return createSquare();
+            case 2:
+                return createRectangle();
+            case 3:
+                return createRightTriangle();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return createIsoscelesTrapezoid();
+    }
+
+    private int getRandomFigure() {
+        return random.nextInt(figureCount);
+    }
+
+    private Circle createCircle() {
+        return new Circle(colorSupplier.getRandomColor(), random.nextInt(defalteSide));
```

Comment:
According the common mistakes "Don't pass expressions to the constructor. " 
Let's introduce appropriate variables for `colorSupplier.getRandomColor()`, `random.nextInt(defalteSide)`
Please fix here and similar cases below.